core/issuers.js

```diff
@@ -17,30 +17,38 @@ Description must be truthy: ${description}`;
 
   // assetSrc is a purse or payment. Return a fresh payment.  One internal
   // function used for both cases, since they are so similar.
-  function takePayment(assetSrc, isPurse, paymentAmount, unsafePaymentName) {
+  function takePayment(
+    assetSrc,
+    srcController,
+    paymentAmount,
+    unsafePaymentName,
+  ) {
     const paymentName = `${unsafePaymentName}`;
     paymentAmount = assay.coerce(paymentAmount);
-    const oldSrcAmount = mintController.getAmount(assetSrc, isPurse);
+    const oldSrcAmount = srcController.getAmount(assetSrc);
```

Comment:
You're right that we don't need it in this call, but we do need it down below in the same function, and using srcController here will fail fast if it is the wrong asset type, so I'm kind of leaning towards this. What do you think?

Reply:
AFAICT, the fail-fast here is required. The key to assaying the alleged src parameter is that it is found as a key in the appropriate encapsulated table, where only genuine srcs of that kind are found.

If you did instead

```js
const oldSrcAmount = assetSrc.getBalance();
```

then Alice could use a misbehaving payment to seem to pay Bob without actually paying him.

Reply:
Btw, I think like the approach you took, or using different controller types rather than the `isPurse` flag. However, I would like for both controllers to be internal mechanism only, not exposed as API that needs documenting. If they are exposed, making two of them increases the API surface yet more.

These are points to discuss later, after this PR is merged.

Reply:
Hi Mark, what do you mean by internal mechanisms vs exposed as API? Does something being in a separate module file make it exposed as an API?

Reply:
Can one create a useful new mint without having to think about mintControllers. I see now that my concern is likely already satisfied enough. Double checking:
   * For simple uses, the caller of makeMint would just have this argument position default, in which case `makeBasicMintController` would be chosen.
   * `makeBasicMintController` is fine for all normal uses that do not need a pre-emption hierarchy.
   * Other than this parameter position, mintControllers are not externally visible, or even mentioned, anywhere the API surface we do expose.

I am not concerned about the separate file. But one thing we should keep in mind as we do safe module and package support for SES is that a package should be able to encapsulate internal modules. I don't know that we would choose to encapsulate `./mintController`, but it will be good to have that choice!